finetrainers/trainer.py

```diff
@@ -660,12 +683,21 @@ def train(self) -> None:
                         text_conditions = batch["text_conditions"]
                         latent_conditions["latents"] = DiagonalGaussianDistribution(
                             latent_conditions["latents"]
-                        ).sample(generator)
-                        if "post_latent_preparation" in self.model_config.keys():
-                            latent_conditions = self.model_config["post_latent_preparation"](**latent_conditions)
-                        align_device_and_dtype(latent_conditions, accelerator.device, weight_dtype)
-                        align_device_and_dtype(text_conditions, accelerator.device, weight_dtype)
-                        batch_size = latent_conditions["latents"].shape[0]
+                        ).sample(torch.Generator(self.state.accelerator.device).manual_seed(self.args.seed))
+                        if is_cog:
+                            if not cog_vae_config["invert_scale_latents"]:
+                                latent_conditions["latents"] = latent_conditions["latents"] * cog_vae_config["scaling_factor"]
+                            else:
+                                latent_conditions["latents"] = 1 / cog_vae_config["scaling_factor"] * latent_conditions["latents"]
+                        
+                    if is_cog:
+                        latent_conditions["latents"] = self._pad_frames(latents=latent_conditions["latents"], denoiser_config=denoiser_config)
+                    
+                    if "post_latent_preparation" in self.model_config.keys():
```

Comment:
This method is probably named a bit incorrectly. It should only be called when precomputation is used. I will take it up and do a refactor